backend/src/app.ts

```diff
@@ -65,9 +66,17 @@ try {
     projectCollection = database.collection<Project>("projects");
     converterCollection = database.collection<Converter>("converters");
 
-    // unique constraint on provider and user id from provider
+    const cursor = database.listCollections({ name: "users" });
+    let created = true;
+
+    // if users doesn't exist set created to true
+    if ((await cursor.toArray()).length != 0) created = false;
+
     userCollection = database.collection<User>("users");
-    await userCollection.createIndex({ provider: 1, id: 1 }, { unique: true });
+    if (created) {
+        // unique constraint on provider and user id from provider
+        await userCollection.createIndex({ provider: 1, _id: 1 }, { unique: true });
```

Comment:
Wtedy zastanawiałem się, czy chcemy mieć to polecenie w kodzie czy nie. Samo to polecenie `await userCollection.createIndex({ provider: 1, _id: 1 }, { unique: true });` może zostać, ale usunąłbym tą całą otoczkę związaną ze sprawdzaniem czy ta kolekcja istnieje czy nie, bo to i tak jest sprawdzane wewnątrz tej funkcji createIndex.